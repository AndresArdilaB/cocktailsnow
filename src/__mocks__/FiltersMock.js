const FiltersMock = [
  { label: 'Categories', param: 'category', items: [{ value: 'Test', key: 'test' }] },
  { label: 'Glasses', param: 'glass', items: [{ value: 'Test', key: 'test' }] },
  { label: 'Ingredients', param: 'ingredient', items: [{ value: 'Test', key: 'test' }] },
  { label: 'Type', param: 'type', items: [{ value: 'Test', key: 'test' }] },
];

export default FiltersMock;