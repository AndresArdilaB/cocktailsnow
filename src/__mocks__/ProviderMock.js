import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { testState } from '../frontend/initialState';
import reducer from '../frontend/reducers';

const store = createStore(reducer, testState);
const history = createBrowserHistory();

const ProviderMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>
      {children}
    </Router>
  </Provider>
);

export default ProviderMock;
