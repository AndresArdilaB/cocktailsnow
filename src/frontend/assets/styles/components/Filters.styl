.filters__main
  display: flex
  align-items: center
  justify-content: center
  flex-direction: column
  height: 250px
  @media screen and (max-width: 1100px)
    height: 370px
  @media screen and (max-width: 600px)
    height: 570px

.filters__title
  color: white
  font-size: 18px
  padding: 20px 0px

.filters
  display: flex
  flex-direction: row
  @media screen and (max-width: 1100px)
    display: grid
    grid-gap: 0.5rem
    grid-template-columns: 260px 260px 
  @media screen and (max-width: 600px)
    display: flex
    flex-direction: column

.search
  width: 1065px
  background-color: rgba(255, 255, 255, 0.1)
  border: 2px solid white
  border-radius: 25px
  color: white
  font-family: "Muli", sans-serif
  font-size: 16px
  height: 50px
  outline: none
  padding: 0px 20px
  @media screen and (max-width: 1100px)
    width: 510px;
  @media screen and (max-width: 600px)
    width: 250px

.order_container
  margin-top: 20px

.order_button
  background-color: rgba(255, 255, 255, 0.1)
  border: 2px solid white
  border-radius: 25px
  padding: 5px 10px
  margin: 0px 4px
  cursor: pointer

::placeholder
  color: white
