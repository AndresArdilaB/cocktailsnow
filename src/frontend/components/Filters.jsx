import React from 'react';
import { useHistory } from 'react-router-dom';
import { connect } from 'react-redux';
import { filterByName, orderByName } from '../actions';
import DropDown from './DropDown';
import '../assets/styles/components/Filters.styl';

const Filters = ({ filters, isHome, filterByName, orderByName }) => {
  const history = useHistory();

  const setFilter = (key, param) => {
    history.push(`/results/${param}/${key.target.value}`);
  };

  const handleFilterByName = (text) => {
    filterByName(text.target.value);
  };

  const handleOrder = (e, type) => {
    e.preventDefault();
    orderByName(type);
  };

  const orderButton = (type) => (
    <span
      className='order_button'
      onClick={(e) => handleOrder(e, type)}
      role='button'
      tabIndex={0}
    >
      {type === 'asc' ? 'Asc' : 'Desc'}
    </span>
  );

  return (
    <section className='filters__main'>
      <div className='filters__title'>
        Find your cocktail by some of this parameters
      </div>
      <section className='filters'>
        {
          //Aqui recibimos el arreglo que enviamos al estado con los filtros
          //Y los podemos generar con un map para evitar repetir codigo llamando
          //DropDown cada que queramos crear un filtro
          filters.map((d, i) => (
            <DropDown
              key={i.toString()}
              items={d.items}
              label={d.label}
              param={d.param}
              callBack={setFilter}
            />
          ))
        }
      </section>
      {
        !isHome && <input type='text' className='search' placeholder='Find by name in this list...' onChange={handleFilterByName} />
      }

      {
        !isHome && (
          <div
            className='order_container'
          >
            Order by name:
            {orderButton('asc')}
            |
            {orderButton('desc')}
          </div>
        )
      }
    </section>
  );
};

const mapDispatchToProps = {
  filterByName,
  orderByName,
};

export default connect(null, mapDispatchToProps)(Filters);
