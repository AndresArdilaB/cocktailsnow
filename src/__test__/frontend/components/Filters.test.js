/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../../__mocks__/ProviderMock';
import FiltersMock from '../../../__mocks__/FiltersMock';
import Filters from '../../../frontend/components/Filters';

describe('<Filters />', () => {

  test('Component render', () => {
    const filters = shallow(
      <ProviderMock>
        <Filters />
      </ProviderMock>,
    );

    expect(filters.length).toEqual(1);
  });

});

describe('Filters Snapshot', () => {
  test('Testing render component UI', () => {
    const filter = create(
      <ProviderMock>
        <Filters
          filters={FiltersMock}
          isHome={true}
          filterByName={() => { }}
          orderByName={() => { }}
        />
      </ProviderMock>,
    );

    expect(filter.toJSON()).toMatchSnapshot();
  });
});
