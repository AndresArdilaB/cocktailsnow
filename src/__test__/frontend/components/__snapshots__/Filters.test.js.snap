// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Snapshot Testing render component UI 1`] = `
<section
  className="filters__main"
>
  <div
    className="filters__title"
  >
    Find your cocktail by some of this parameters
  </div>
  <section
    className="filters"
  >
    <div
      className="container"
    >
      <select
        className="select"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Categories
        </option>
        <option
          id="test"
          value="test"
        >
          Test
        </option>
      </select>
    </div>
    <div
      className="container"
    >
      <select
        className="select"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Glasses
        </option>
        <option
          id="test"
          value="test"
        >
          Test
        </option>
      </select>
    </div>
    <div
      className="container"
    >
      <select
        className="select"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Ingredients
        </option>
        <option
          id="test"
          value="test"
        >
          Test
        </option>
      </select>
    </div>
    <div
      className="container"
    >
      <select
        className="select"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Type
        </option>
        <option
          id="test"
          value="test"
        >
          Test
        </option>
      </select>
    </div>
  </section>
</section>
`;
